# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/brush_blend.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/brush_blend.glsl
--- a/gfx/wr/webrender/res/brush_blend.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/brush_blend.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -8,16 +8,16 @@
 #include shared,prim_shared,brush,blend
 
 // Interpolated UV coordinates to sample.
-varying vec2 v_uv;
+varying highp vec2 v_uv;
 
 // Normalized bounds of the source image in the texture, adjusted to avoid
 // sampling artifacts.
-flat varying vec4 v_uv_sample_bounds;
+flat varying highp vec4 v_uv_sample_bounds;
 
 // x: Flag to allow perspective interpolation of UV.
 // y: Filter-dependent "amount" parameter.
 // Packed in to a vector to work around bug 1630356.
-flat varying vec2 v_perspective_amount;
+flat varying mediump vec2 v_perspective_amount;
 #define v_perspective v_perspective_amount.x
 #define v_amount v_perspective_amount.y
 
@@ -30,12 +30,12 @@
 #define v_op v_op_table_address_vec.x
 #define v_table_address v_op_table_address_vec.y
 
-flat varying mat4 v_color_mat;
+flat varying mediump mat4 v_color_mat;
 // The function to use for each component of a component transfer filter. Using a int[4]
 // or ivec4 (with each element or component containing the function for each component) has
 // ran in to bugs 1695912 and 1731758, so instead use a vec4 and cast the values to/from floats.
-flat varying vec4 v_funcs;
-flat varying vec4 v_color_offset;
+flat varying mediump vec4 v_funcs;
+flat varying mediump vec4 v_color_offset;
 
 #ifdef WR_VERTEX_SHADER
 void brush_vs(